src/app/ui/context_bar.cpp

```diff
@@ -2167,7 +2167,7 @@ void ContextBar::updateForTool(tools::Tool* tool)
   // Show/Hide fields
   m_zoomButtons->setVisible(needZoomButtons(tool));
   m_brushBack->setVisible(supportOpacity && hasImageBrush && !withDithering);
-  m_brushType->setVisible(supportOpacity && (!isFloodfill || (isFloodfill && hasImageBrush && !withDithering)));
+  m_brushType->setVisible(supportOpacity && (!isFloodfill || (isFloodfill && !withDithering)));
```

Comment:
To avoid confusion for the user, I'd leave only this change and keep the brush popup just as it looks when other tools are selected (and avoid hiding elements).

Reply:
Despite being a much easier fix I have the need of asking: What about disabling the brushes instead of hiding them?

Reply:
I thought about the same but I don't know if it worth it. Just in case we need a way to go back to a plain brush to fill with solid colors (not sure if that was the case with this patch, the regular/solid brush types must be visible).

There are plans to redesign this popup in the future anyway (#3219) so probably doesn't worth the effort.